backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/incremental/updater/ContainmentUpdater.java

```diff
@@ -158,24 +158,11 @@ private void updateNodeListItem(NodeLayoutData nodeLayoutData) {
     }
 
     private void updateBottomRight(IContainerLayoutData container) {
-        double width = container.getSize().getWidth();
-        double height = container.getSize().getHeight();
-        double maxWidth = LayoutOptionValues.MIN_WIDTH_CONSTRAINT;
-        double maxHeight = LayoutOptionValues.MIN_HEIGHT_CONSTRAINT;
+        Size contentSize = this.computeContentSize(container);
 
-        if (this.isLabelContained(container)) {
-            LabelLayoutData label = ((NodeLayoutData) container).getLabel();
-            maxWidth = Math.max(maxWidth, label.getTextBounds().getSize().getWidth() + this.getNodeLabelPaddingWidth(container));
-            maxHeight = Math.max(maxHeight, label.getTextBounds().getSize().getHeight() + this.getNodeLabelPaddingHeight(container));
-        }
-
-        for (NodeLayoutData child : container.getChildrenNodes()) {
-            maxWidth = Math.max(maxWidth, child.getPosition().getX() + child.getSize().getWidth() + this.getPaddingWidth(container));
-            maxHeight = Math.max(maxHeight, child.getPosition().getY() + child.getSize().getHeight() + this.getPaddingHeight(container));
-        }
-
-        if (maxWidth != width || maxHeight != height) {
-            container.setSize(Size.of(maxWidth, maxHeight));
+        // Only update the conttainer's size if it is too small for its content.
+        if (container.getSize().getWidth() < contentSize.getWidth() || container.getSize().getHeight() < contentSize.getHeight()) {
+            container.setSize(contentSize);
```

Comment:
I think a NodeList should be resized if the sum of its NodeListItems size is smaller than the NodeList size.